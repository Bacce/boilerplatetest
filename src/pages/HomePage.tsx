
import { Col, GridContainer, Hidden, MainContainer } from '../components/Layout';
import { languagesEnum } from "../i18n";
import { useTranslation } from 'react-i18next';
import { Placeholder } from '../components/Placeholder';
import { Spacer } from '../components/Spacer';
import { FeatureToggle } from "../components/FeatureToggle";
import { featuresEnum } from "../enums/featuresEnum";

export const HomePage = () => {
  const { t, i18n } = useTranslation();

  return (
  <MainContainer>
    <FeatureToggle feature={featuresEnum.FEATURE_1} offTemplate={<div>Feature 1 off</div>}>
      <div>Feature 1 is enabled</div>
    </FeatureToggle>
    <FeatureToggle feature={featuresEnum.FEATURE_2} offTemplate={<div>Feature 2 off</div>}>
      <div>Feature 2 is enabled</div>
    </FeatureToggle>
    <FeatureToggle feature={featuresEnum.FEATURE_3} offTemplate={<div>Feature 3 off</div>}>
      <div>Feature 3 is enabled</div>
    </FeatureToggle>
    <FeatureToggle feature={featuresEnum.FEATURE_4} offTemplate={<div>Feature 4 off</div>}>
      <div>Feature 4 is enabled</div>
    </FeatureToggle>

  <GridContainer>
    <Col>
      <p>Language: {i18n && i18n.language}</p>

      <button onClick={()=>i18n.changeLanguage(languagesEnum.esES)}>Spanish</button>
      <button onClick={()=>i18n.changeLanguage(languagesEnum.enGB)}>English</button>
      <p>{t('home.paragraph')}</p>
    </Col>
  </GridContainer>
  <Hidden breakpoints={["sm"]}>
    <p>test only hidden on SM</p>
  </Hidden>
  <GridContainer>
      <Col sm={2} md={4} lg={6}>
        <Spacer size={8} />
        <Placeholder/>
      </Col>
      <Col sm={2} md={4} lg={6}>
        <Placeholder/>
      </Col>
{/* ---- */}
      <Col>
        <Placeholder/>
      </Col>
      <Col>
        <Placeholder/>
      </Col>
      <Col>
        <Placeholder/>
      </Col>
      <Col>
        <Placeholder/>
      </Col>
{/* ---- */}
      <Col>
        <Placeholder/>
      </Col>
      <Col>
        <Placeholder/>
      </Col>
      <Col>
        <Placeholder/>
      </Col>
      <Col>
        <Placeholder/>
      </Col>
{/* ---- */}
      <Col>
        <Placeholder/>
      </Col>
      <Col>
        <Placeholder/>
      </Col>
      <Col>
        <Placeholder/>
      </Col>
      <Col>
        <Placeholder/>
      </Col>
  </GridContainer>
  </MainContainer>);
}