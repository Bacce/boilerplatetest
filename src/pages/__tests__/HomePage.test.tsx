import { render } from '@testing-library/react';
import { HomePage } from '../HomePage';

describe('HomePage Component', () => {    
    beforeEach(()=>{
        // https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
              matches: false,
              media: query,
              onchange: null,
              addListener: jest.fn(), // deprecated
              removeListener: jest.fn(), // deprecated
              addEventListener: jest.fn(),
              removeEventListener: jest.fn(),
              dispatchEvent: jest.fn(),
            })),
          });
    });

    it('should render', ()=>{
        const view = render(<HomePage/>);
        
        expect(view.container).toMatchSnapshot();
    });
});
