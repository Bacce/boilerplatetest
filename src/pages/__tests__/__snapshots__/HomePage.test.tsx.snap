// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Component should render 1`] = `
<div>
  <div
    class="main-container"
  >
    <div>
      Feature 1 is enabled
    </div>
    <div>
      Feature 2 is enabled
    </div>
    <div>
      Feature 3 is enabled
    </div>
    <div>
      Feature 4 is enabled
    </div>
    <div
      class="grid-container"
    >
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <p>
          Language: 
          enGB
        </p>
        <button>
          Spanish
        </button>
        <button>
          English
        </button>
        <p>
          Test paragraph in English
        </p>
      </div>
    </div>
    <div
      class="grid-container"
    >
      <div
        class="col-sm-2 col-md-4 col-lg-6"
      >
        <div
          style="width: 8px; height: 8px;"
        />
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-2 col-md-4 col-lg-6"
      >
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <div
          class="container"
        />
      </div>
      <div
        class="col-sm-1 col-md-1 col-lg-1"
      >
        <div
          class="container"
        />
      </div>
    </div>
  </div>
</div>
`;
